lib/provider.py

```diff
@@ -97,7 +98,26 @@ def get_factor(self, seeds_factor=4, default_seeds=0, leeches_factor=1, default_
         return max(seeds * seeds_factor + leeches * leeches_factor, 1) * resolution
 
 
-def perform_search(search_type, data):
+def include_result(result):
```

Comment:
This method seems to be ok, however I would like to suggest 2 improvements:
1. We should avoid accessing protected fileds outside of the class implementation. As such I would suggest moving this method to the Result implementation or, if you prefer keeping it outside the Result implementation, do getters (or `@property`) for those protected fields.
2. Not sure about this one, but probably we should cache these settings as doing get_boolean_setting(...) thousands of times might be expensive. A simple yet effective solution for this would be implementing a cache using a dict, where it would first try getting the setting from the dict and if it is not there, it would get it from kodi.

Reply:
On an additional note, we could simplify this to a single statement:
```python
def include_result(result):
    return ((not get_boolean_setting("require_size") or bool(result._size)) and
            (not get_boolean_setting("require_seeds") or bool(result.seeds)) and
            (not get_boolean_setting("require_resolution") or get_boolean_setting(
                "include_{}".format(result._resolution.name.lower()))) and
            (not get_boolean_setting("require_release_type") or get_boolean_setting(
                "include_{}".format(result._release.name.lower()))))
```

Sorry for being picky :D

Reply:
I moved the `include_result` method in the `Result` class. What do you prefer for settings caching?

```python
def perform_search(search_types, data):
    settings = {}

    for item in ["size", "seeds", "resolution", "release_type"]:
        setting = "require_{}".format(item)
        settings[setting] = get_boolean_setting(setting)

    for item in (Resolution.values + ReleaseType.values):
        setting = "include_{}".format(item.name.lower())
        settings[setting] = get_boolean_setting(setting)
    # ...
```

```python
class CachedSettings(object):
    def __init__(self):
        self._cache = {}

    def get_boolean(self, key):
        if key not in self._cache:
          self._cache[key] = get_boolean_setting(key)
        return self._cache[key]


def perform_search(search_types, data):
    settings = CachedSettings()
    # ...
```